### PR TITLE
Basic Types - Instruction

### PR DESCRIPTION
**Pull request by shakram02**
-   Code for the instruction struct that will be used with pass 2 ( flag resolution )
-   Tests for *most* of code cases

### CODE REVIEW

**Comment by oddcoder** [comment]
pls dont merge with master ... just rebase next time like this
```
git rebase upstream master
```

File: src/lib/pass_two/tests.rs [hunk 1]
```diff
@@ -1,12 +1,11 @@
 #[cfg(test)]
 mod tests {
-    use super::*;
+    //    use super::*;
 
     #[test]
     #[should_panic]
     fn it_panics() {
-        panic!("diee")
+        println!("I'll die now");
```
Comment (oddcoder) [inline src/lib/pass_two/tests.rs]: what does this test do ?

**Review by oddcoder — COMMENTED** [review]
To be continued

File: src/lib/pass_two/tests.rs (hunk 1 above)
Comment (shakram02) [reply src/lib/pass_two/tests.rs]: confirms that tests run correctly ( I used it in the beginning of time )

Comment (oddcoder) [reply src/lib/pass_two/tests.rs]: you can get rid of this now ?

**Comment by oddcoder** [comment]
merged thanks!

**Pull request closed without merging**